sejda-core/src/main/java/org/sejda/core/support/io/DefaultSingleOutputWriter.java

```diff
@@ -20,25 +20,65 @@
  */
 package org.sejda.core.support.io;
 
-import org.sejda.core.support.io.model.PopulatedFileOutput;
+import static java.util.Objects.isNull;
+import static java.util.Optional.of;
+import static org.sejda.core.support.io.OutputWriterHelper.moveFile;
+import static org.sejda.model.output.ExistingOutputPolicy.FAIL;
+import static org.sejda.model.output.ExistingOutputPolicy.SKIP;
+
+import java.io.File;
+import java.io.IOException;
+
+import org.sejda.model.output.DirectoryTaskOutput;
 import org.sejda.model.output.ExistingOutputPolicy;
+import org.sejda.model.output.FileOrDirectoryTaskOutput;
+import org.sejda.model.output.FileTaskOutput;
 import org.sejda.model.task.TaskExecutionContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 /**
  * Single writer default implementation.
  * 
  * @author Andrea Vacondio
  * 
  */
-class DefaultSingleOutputWriter extends BaseOutputWriter implements SingleOutputWriter {
+class DefaultSingleOutputWriter implements SingleOutputWriter {
+
+    private static final Logger LOG = LoggerFactory.getLogger(DefaultSingleOutputWriter.class);
+
+    private File taskOutput;
+    private final ExistingOutputPolicy existingOutputPolicy;
+    private final TaskExecutionContext executionContext;
 
     DefaultSingleOutputWriter(ExistingOutputPolicy existingOutputPolicy, TaskExecutionContext executionContext) {
-        super(existingOutputPolicy, executionContext);
+        this.existingOutputPolicy = of(existingOutputPolicy).filter(p -> p != SKIP).orElseGet(() -> {
+            LOG.debug("Cannot use {} output policy for single output, replaced with {}", SKIP, FAIL);
```

Comment:
I think it's not new in this branch, I moved it from another place. The task generates 1 file and we tell it, in case of existence, to skip that one file that it's suppose to generate so it becomes a task that doesn't produce any output, I'd say it's a failure.. or do you think we should honor the SKIP and succeed without any output?